tests/test___main__.py

```diff
@@ -6,40 +6,74 @@
 from sqlrunner.__main__ import entrypoint
 
 
-def test_entrypoint(monkeypatch, tmp_path, dsn):
+@pytest.fixture
+def input_file(monkeypatch, tmp_path):
     monkeypatch.chdir(tmp_path)
-    pathlib.Path("input.sql").write_text(
-        f"-- {T1OOS_TABLE} intentionally not excluded\nSELECT 1 AS patient_id",
+
+    path = pathlib.Path("input.sql")
+    path.write_text(
+        f"-- {T1OOS_TABLE} intentionally not excluded\nSELECT 1 AS Patient_ID",
         "utf-8",
     )
-    monkeypatch.setattr(
-        "sys.argv",
-        [
-            "__main__",
-            "--output",
-            "output.csv",
-            "--log-file",
-            "log.json",
-            "input.sql",
-        ],
-    )
+    return str(path)
+
+
+@pytest.mark.parametrize("pass_dummy_data_file", [False, True])
+def test_entrypoint_with_dsn(
+    monkeypatch, tmp_path, dsn, pass_dummy_data_file, input_file
+):
+    monkeypatch.chdir(tmp_path)
     monkeypatch.setattr("os.environ", {"DATABASE_URL": dsn})
+
+    argv = ["__main__", "--output", "output.csv", "--log-file", "log.json"]
+
+    if pass_dummy_data_file:
+        pathlib.Path("dummy_data.csv").touch()  # Notice the dummy data file is empty
+        argv.extend(["--dummy-data-file", "dummy_data.csv"])
+
+    argv.append(input_file)
+
+    monkeypatch.setattr("sys.argv", argv)
+
     entrypoint()
-    assert pathlib.Path("output.csv").read_text("utf-8") == "patient_id\n1\n"
+
+    # The dummy data file is empty, so we can be confident that the output file contains
+    # the result of executing the query in the input file against the database.
+    assert pathlib.Path("output.csv").read_text("utf-8") == "Patient_ID\n1\n"
+
+    # We test what is being logged elsewhere.
     assert pathlib.Path("log.json").exists()
 
 
-def test_entrypoint_with_missing_args(monkeypatch):
+def test_entrypoint_without_dsn_with_dummy_data_file(monkeypatch, tmp_path, input_file):
+    monkeypatch.chdir(tmp_path)
+
+    # The dummy data file and the query in the input file differ. Although this would be
+    # a bug in a study, it allows us to test that the output file doesn't contain the
+    # result of executing the query in the input file against the database.
+    pathlib.Path("dummy_data.csv").write_text("Sex\nF\n", "utf-8")
```

Comment:
The encoding is unnecessary here (utf-8 is the default) and is a little bit distracting, so I'd remove it.

Reply:
The value of the `encoding` argument is `"utf-8"` on macOS and Linux, but I'm unsure about Windows: the default is platform-dependant ([`locale.getpreferredencoding`](https://docs.python.org/3.10/library/locale.html#locale.getpreferredencoding)). We're explicit elsewhere, so for consistency I was explicit here too.